csharp/src/Ice/Communicator.cs

```diff
@@ -1229,6 +1240,14 @@ public IceInternal.Timer Timer()
             }
         }
 
+        // Registers an endpoint factory.
+        // TODO: make public and add Ice prefix when removing ITransportPluginFacade.
+        internal void AddEndpointFactory(IEndpointFactory factory)
+        {
+            _typeToEndpointFactory.Add(factory.Type(), factory);
+            _transportToEndpointFactory.Add(factory.Transport(), factory);
```

Comment:
This can throw `ArgumentException` if a factory for the type/transport is already registered, 

I guess it is fine, previous code just have an assert for the type case, and didn't check the transport, maybe we should catch the exception and wrap it in another `ArgumentException` with a more specific message and keep the original as the inner exception

Reply:
I think it's fine as is; it's all Ice-internal anyway.